### PR TITLE
Added subgroups and group_edit methods

### PR DESCRIPTION
**Pull request by Rubemlrm**
Hi there,

First of all thanks for the hard work, this week I've tried use the gem and i've found that are missing some api methods at groups.rb, so i take some time to add the following methods:

- List Groups Subgroups based on the group ID we send on the API call (https://docs.gitlab.com/ce/api/groups.html#list-a-groups-s-subgroups)
- Edit group method, based on the group ID we send on the API call (https://docs.gitlab.com/ce/api/groups.html#update-group)

Besides this, i've added the fixtures and specs for testing this new 2 methods.

### CODE REVIEW

File: lib/gitlab/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module Gitlab
-  VERSION = '4.2.0'.freeze
+  VERSION = '4.2.1'.freeze
```
Comment (NARKOZ) [inline lib/gitlab/version.rb]: Please don't change this file

File: lib/gitlab/client/groups.rb [hunk 2]
```diff
@@ -152,5 +152,42 @@ def group_search(search, options={})
     def group_projects(id, options={})
       get("/groups/#{id}/projects", query: options)
     end
+
+    # Get a list of subgroups under a group
+    # @example
+    #   Gitlab.group_projects(1)
```
Comment (NARKOZ) [inline lib/gitlab/client/groups.rb]: `Gitlab.group_subgroups(1)`

**Comment by Rubemlrm** [comment]
hi there, sorry for the last reply. I've submitted the requested changes.

**Comment by NARKOZ** [comment]
Thank you.

**Pull request merged**